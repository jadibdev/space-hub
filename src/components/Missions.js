import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getMissions, joinMission, cancelMission } from '../redux/missions/missionSlice';

const Missions = () => {
  const dispatch = useDispatch();
  const { missions } = useSelector((state) => state.missions);
  const { joinedMissions, reserved } = useSelector((state) => state.missions);

  const handleClick = (id) => {
    if (!joinedMissions.includes(id)) {
      dispatch(joinMission(id));
    } else {
      dispatch(cancelMission(id));
    }
  };

  useEffect(() => {
    dispatch(getMissions());
  }, [dispatch]);
  return (
    <div>
      <table>
        <thead>
          <tr>
            <th>Missions</th>
            <th>Description</th>
            <th>Status</th>
            <th>-</th>
          </tr>
        </thead>
        <tbody>
          {missions.map((mission) => (
            <tr key={mission.mission_id}>
              <td>{mission.mission_name}</td>
              <td>{mission.description}</td>
              <td className={joinedMissions.includes(mission.mission_id) ? 'member' : 'non-member'}>{joinedMissions.includes(mission.mission_id) ? 'Active member' : 'Not a member'}</td>
              <td>
                {reserved ? <button className={joinedMissions.includes(mission.mission_id) ? 'cancel' : 'join'} id={mission.mission_id} type="button" onClick={() => handleClick(mission.mission_id)}>{joinedMissions.includes(mission.mission_id) ? 'Cancel mission' : 'Join mission'}</button>
                  : <button className={joinedMissions.includes(mission.mission_id) ? 'cancel' : 'join'} id={mission.mission_id} type="button" onClick={() => handleClick(mission.mission_id)}>{joinedMissions.includes(mission.mission_id) ? 'Cancel mission' : 'Join mission'}</button>}
              </td>
            </tr>

          ))}
        </tbody>
      </table>
    </div>
  );
};
export default Missions;
