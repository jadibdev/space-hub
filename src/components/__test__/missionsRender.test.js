import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import Missions from '../Missions';

it('Missions component renders successfully!', () => {
  const tree = renderer
    .create(<Provider store={store}><Missions /></Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
