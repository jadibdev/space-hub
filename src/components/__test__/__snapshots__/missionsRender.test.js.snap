// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions component renders successfully! 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th>
          Item name
        </th>
        <th>
          Item description
        </th>
        <th>
          Status
        </th>
        <th>
          -
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
